### PR TITLE
Expand automated testing for main functionality

### LINKED ISSUE #79: Add playwright tests to validate dropdown options
- [x] Use playwright to test every combination of datatype, from, and to values and ensure they are all valid
- [x] Test feedback forms

### PR DESCRIPTION
**Pull request by alexbainter**
Resolves #79.

- Adds end-to-end testing with Playwright for the feedback forms on the index page, article list page (when there are multiple matching results for the user's query), and the article page.
- Updates the index page tests to ensure there's at least one datatype, and then programmatically check each datatype's source and destination combinations to ensure they are all valid.
  - I know we discussed that we might not need to do this, but I was basically done with it by the time we had that conversation :sweat_smile:. I do think there's value here because this is the primary functionality of the site and it requires client-side JS, so testing it in a browser environment makes sense.
  - Even with this in place, all the tests run in [a little over 30 seconds](https://github.com/dtinit/portmap/actions/runs/8974432720/job/24646805665?pr=89#step:8:25)

### CODE REVIEW

File: tests/e2e/test_article.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        page = self.context.new_page()
+        page.goto('/articles/' + TEST_ARTICLE_NAME + '/')
+        page.locator('form#query_form input[value=sad]').click()
+        page.get_by_text('Give feedback').click();
+        expect(page).to_have_url('/articles/testarticle.md/feedback')
+        feedbackCollection = Feedback.objects.all()
+        assert len(feedbackCollection) == 1
+        newFeedback = feedbackCollection.first()
+        assert newFeedback.article.name == TEST_ARTICLE_NAME
+        assert newFeedback.reaction == 'sad'
```
Comment (alexbainter) [inline tests/e2e/test_article.py:66]: I tried to [parametrize this](https://docs.pytest.org/en/latest/how-to/parametrize.html) with "happy" and "sad" but it didn't work

File: tests/e2e/test_article.py [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+import pytest
+from django.contrib.staticfiles.testing import StaticLiveServerTestCase
+from playwright.sync_api import expect, sync_playwright
+from portmap.core.models import Article, Feedback
+
+# Tests for articles
+
+TEST_ARTICLE_NAME = 'testarticle.md'
+
+def create_test_article():
```
Comment (lisad) [inline tests/e2e/test_article.py:12]: A perfect use case for a Factory

File: tests/e2e/test_find_articles.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    # When there are multiple matching articles,
+    # the user is presented a list to choose from and a feedback form.
+    @pytest.mark.django_db
+    def test_usecase_feedback_form(self):
+        mock_datatype = 'Mock Datatype'
+        mock_source = 'Mock Source'
+        mock_destination = 'Mock Destination'
+        mock_explanation = 'TEST FEEDBACK FROM PLAYWRIGHT'
+
+        Article.objects.create(
```
Comment (lisad) [inline tests/e2e/test_find_articles.py:33]: if we had an article factory, this test could call it twice :)

**Review by lisad — APPROVED** [review]

**Comment by alexbainter** [comment]
Updated to use a factory! https://github.com/dtinit/portmap/pull/89/commits/05c1ca2c3aff91416a0b9e5975b8f8b50d57503e

**Pull request merged**